frontend/Dockerfile

```diff
@@ -0,0 +1,14 @@
+# Stage 1: Compile and Build angular codebase
+
+# Use official node image as the base image
+FROM node:latest as build
+
+# Set the working directory
+COPY . /code
+WORKDIR /code
+
+# Install all the dependencies
+RUN npm install
+
+# Generate the build of the application
+RUN npm run build
```

Comment:
Does this dockerfile want an entrypoint or `cmd` so that someone can `docker run` just in this directory? Or is it only ever to be used with the docker-compose setup in the next directory up? If so, perhaps a comment at the end of this file to that effect?

Reply:
Since neither Dockerfile has an `ENTRYPOINT` or `CMD`, my preference would be to not document this, and let it be the convention for this project that Dockerfiles are just used for `docker-compose`. Feel free to push back on this; it's not a _strong_ preference.